apps/prairielearn/src/pages/instructorStudents/instructorStudents.tsx

```diff
@@ -0,0 +1,84 @@
+import { Router } from 'express';
+import asyncHandler from 'express-async-handler';
+
+import { loadSqlEquiv, queryRows } from '@prairielearn/postgres';
+
+import { InsufficientCoursePermissionsCardPage } from '../../components/InsufficientCoursePermissionsCard.js';
+import { PageLayout } from '../../components/PageLayout.html.js';
+import { getCourseInstanceContext, getPageContext } from '../../lib/client/page-context.js';
+import { getCourseOwners } from '../../lib/course.js';
+import { Hydrate } from '../../lib/preact.js';
+import { getUrl } from '../../lib/url.js';
+
+import { InstructorStudents } from './instructorStudents.html.js';
+import { StudentQuerySchema, type StudentRow } from './instructorStudents.shared.js';
+
+const router = Router();
+const sql = loadSqlEquiv(import.meta.url);
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const pageContext = getPageContext(res.locals);
+    const { course_instance, course } = getCourseInstanceContext(res.locals, 'instructor');
+
+    const search = getUrl(req).search;
+
+    const hasPermission = pageContext.authz_data.has_course_instance_permission_view;
+    if (!hasPermission) {
+      const courseOwners = await getCourseOwners(course.id);
+      res.status(403).send(
+        InsufficientCoursePermissionsCardPage({
+          resLocals: res.locals,
+          courseOwners,
+          pageTitle: 'Students',
+          requiredPermissions: 'Instructor',
+        }),
+      );
+    }
+
+    const students: StudentRow[] = hasPermission
```

Comment:
The `hasPermission` check shouldn't be needed here.

Reply:
Fixed in 1be0e4a6aeddafe6b0ea9fc3ee72f8e534369743 / 6bfdc21e358d030e70cbbb0b61c07a8022d3ed17.